Volume/Views/WeeklyDebriefView.swift

```diff
@@ -6,112 +6,251 @@
+                fetchInfoFor(debrief: debrief)
+            }
+    }
+    
+    private func fetchInfoFor(debrief: WeeklyDebrief) {
+        
+        var debriefDisplayInfo = WeeklyDebriefDisplayInfo(
+            creationDate: debrief.creationDate,
+            expirationDate: debrief.expirationDate,
+            numShoutouts: debrief.numShoutouts,
+            numReadArticles: debrief.numReadArticles,
+            numBookmarkedArticles: debrief.numBookmarkedArticles,
+            readArticles: [],
+            randomArticles: [])
+        
+        func asyncPopulateDisplayInfo(recievedArticles: Int = 0, ids: [ArticleID], readNotRandom: Bool, completion: @escaping () -> ()) {
+            if recievedArticles == ids.count {
+                completion()
+                return
+            }
+            
+            cancellableArticleQuery = Network.shared.publisher(for: GetArticleByIdQuery(id: ids[recievedArticles]))
+                .sink { completion in
+                    if case let .failure(error) = completion {
+                        print(error.localizedDescription)
+                    }
+                }
+                receiveValue: { articleFields in
+                    let article = Article(from: articleFields.article!.fragments.articleFields)
+                    
+                    if readNotRandom {
+                        debriefDisplayInfo.readArticles.append(article)
+                    } else {
+                        debriefDisplayInfo.randomArticles.append(article)
+                    }
+                    
+                    asyncPopulateDisplayInfo(recievedArticles: recievedArticles + 1, ids: ids, readNotRandom: readNotRandom, completion: completion)
+                }
+        }
+        
+        asyncPopulateDisplayInfo(ids: debrief.readArticleIDs, readNotRandom: true) {
+            asyncPopulateDisplayInfo(ids: debrief.randomArticleIDs, readNotRandom: false) {
+                state = .results(debriefDisplayInfo)
+            }
+        }
+        
+    }
+    
+    /**
+     * Creates a date string for the past week
+     */
+    private func getWeekString() -> String {
+        var message = "Your weekly debrief"
+        
+        if let firstDay = weeklyDebrief?.creationDate, let lastDay = weeklyDebrief?.expirationDate {
+            let formatter = DateFormatter()
+            formatter.setLocalizedDateFormatFromTemplate("M/dd")
+            
+            let firstString = formatter.string(from: firstDay)
+            let lastString = formatter.string(from: lastDay)
+            message.append(contentsOf: ", " + firstString + " - " + lastString)
+        }
+        
+        return message
+    }
+    
 }
 
-//struct WeeklyDebriefView_Previews: PreviewProvider {
-//    static var previews: some View {
-//        WeeklyDebriefView()
-//    }
-//}
+extension WeeklyDebriefView {
+    private enum WeeklyDebriefViewState<Result> {
+        case loading, results(Result)
+    }
```

Comment:
Instead of reimplementing this for `WeeklyDebriefView`, consider modifying `MainView.TabState` by adding a `.weeklyDebriefPopup` case & renaming `TabState` to `ScreenState`. Remember to design APIs for the most general-purpose usage.